src_c/math.c

```diff
@@ -2082,6 +2084,40 @@ static PyBufferProcs vector_as_buffer = {
 };
 #endif
 
+static PyObject *
+vector___round__(pgVector *self, PyObject *args)
+{
+    Py_ssize_t i, ndigits;
+    PyObject *o_ndigits = NULL;
+
+    pgVector *ret = _vector_subtype_new(self);
+
+    if (ret == NULL || !PyArg_ParseTuple(args, "|O", &o_ndigits)) {
+        return NULL;
```

Comment:
I see what you're doing here, and it does the make the code shorter, but does it make it harder to read? I don't have a strong opinion just was thinking.

Reply:
I would very much prefer if this was done on 2 separate `if` blocks instead of one. The reason being that in this code, `PyArg_ParseTuple` will be executed regardless of the first check. But if `ret` is `NULL` we are supposed to exit immediately as it is most likely a memory error without calling more python c api functions

Reply:
@ankith26 should it simply `return NULL;` if `ret` is `NULL` or also set an error message?

Reply:
Both PyArg_Parse and vector_subtype_new will set the error themselves, so no need for you to set the error.